src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -25,266 +25,234 @@
+  private def buildVariantMask(variants: Option[PathToVcf], intervals: Option[IntervalList], dict: SAMSequenceDictionary): VariantMask = {
+    this.variants match {
+      case None =>
+        new VariantMask(Iterator.empty, dict)
+      case Some(path) =>
+        val reader = new VCFFileReader(path.toFile, this.intervals.isDefined)
+        intervals match {
+          case None => new VariantMask(reader.iterator(), dict)
+          case Some(i) => new VariantMask(ByIntervalListVariantContextIterator(reader, i), dict)
         }
-        else if (op.consumesReadBases) {
-          readOffset += readOffsetIncrement * elem.getLength
-        }
-      }
     }
   }
-}
-
-/** Computes the error rate using the MD tag. Does not rely on a reference FASTA. */
-private class ErrorRateByReadPositionWithMdCounter(includeInsertions: Boolean, minBaseQuality: Int)
-  extends ErrorRateByReadPositionCounter(includeInsertions=includeInsertions, minBaseQuality=minBaseQuality) {
 
-  protected def updateErrors(rec: SAMRecord, readIndex: Int): Unit  = {
-    // Use the MD tag!
-    val read              = maybeReverseComplement(rec.getReadBases.clone(), rc=rec.getReadNegativeStrandFlag)
-    val quals             = maybeReverse(rec.getBaseQualities.clone(), rc=rec.getReadNegativeStrandFlag)
-    val reference         = maybeReverseComplement(SequenceUtil.makeReferenceFromAlignment(rec, false), rc=rec.getReadNegativeStrandFlag)
-    var readPosition      = 0
+  /** Compute the read number for the record. */
+  def readNumber(rec: SAMRecord): Int = if (!rec.getReadPairedFlag) 0 else if (rec.getFirstOfPairFlag) 1 else 2
+}
 
-    while (readPosition < read.length) {
-      val readBase = read(readPosition)
-      val qual     = quals(readPosition)
-      reference(readPosition) match {
-        case '-'                       => // insertion
-          if (includeInsertions) this.increment(readIndex=readIndex, readOffset=readPosition, qual=qual) // insertion
-        case refBase if '0' != refBase => // not a soft-clip
-          this.incrementIfValidBase(readIndex=readIndex, readOffset=readPosition, readBase=readBase, qual=qual, refBase=refBase)
-        case refBase                   => // ignore
-      }
-      readPosition += 1
-    }
+private class ObsCounter(readNumber: Int, position: Int) {
+  var a_ref_obs: Long  = 0
+  var c_ref_obs: Long  = 0
+  var a_to_c_obs: Long = 0
+  var a_to_g_obs: Long = 0
+  var a_to_t_obs: Long = 0
+  var c_to_a_obs: Long = 0
+  var c_to_g_obs: Long = 0
+  var c_to_t_obs: Long = 0
+
+  def toMetric: ErrorRateByReadPositionMetric = {
+    val totalA = a_ref_obs + a_to_c_obs + a_to_g_obs + a_to_t_obs
+    val totalC = c_ref_obs + c_to_a_obs + c_to_g_obs + c_to_t_obs
+    val total  = totalA + totalC
+
+    new ErrorRateByReadPositionMetric(
+      read_number = readNumber,
+      position    = position,
+      bases_total = total,
+      error_rate  = if (total == 0) 0 else (total - a_ref_obs - c_ref_obs) / total.toDouble,
```

Comment:
We do this so often, and sometimes I forget, perhaps we should put this in an fgbiodef.